flax/linen/module.py

```diff
@@ -725,6 +725,12 @@ def init(self, rngs: Union[PRNGKey, RNGSequences], *args,
            **kwargs) -> VariableDict:
     """Initializes a module method with variables and returns modified variables.
 
+    Calling `jax.jit` on init will make JAX trace through the model and compile
```

Comment:
I would focus here on the laziness aspect (not computing the actual forward pass) and show an example of how to do that. The info here is correct but only relevant if you also have a lazy_init which we currently don't offer

Reply:
Done.